LabController/src/bkr/labcontroller/concurrency.py

```diff
@@ -29,22 +30,27 @@ def _read_from_pipe(f):
     discarding = False
     while True:
         try:
+            gevent.socket.wait_read(f.fileno())
             chunk = f.read(4096)
             if not chunk:
                 break
             if not discarding:
                 chunks.append(chunk)
                 if len(chunks) >= 1000:
-                    logger.error('Too many chunks read from fd %s, '
-                            'child process is running amok?!', f.fileno())
-                    chunks.append('+++ DISCARDED')
+                    logger.error(
+                        "Too many chunks read from fd %s, "
+                        "child process is running amok?!",
+                        f.fileno(),
+                    )
+                    chunks.append(b"+++ DISCARDED")
```

Comment:
Should this have a PY2/PY3 check?  Bytes versus non-bytes?

Reply:
No, this is actually fine. In Python 2, `str` == `bytes`, so I can just use `b''` here, but since the rest of the system needs str everywhere, I will decode it at the end for Python3.